src/clojure/nrepl/transport.clj

```diff
@@ -164,6 +164,14 @@
             (.close in)
             (.close out))))))))
 
+(defn tty-read-msg
```

Comment:
I really don't like returning a function from this, if you have suggestions I'd appreciate it.

Reply:
You can always keep it as a lambda and do some integration test of the transport instead. I guess the only reason you made this a public function is to be able to write that unit test for it.

Reply:
Yeah, I did it so I could demonstrate the change was very focused.  I'll put it back.